velox/exec/benchmarks/OrderByBenchmark.cpp

```diff
@@ -0,0 +1,196 @@
+    const std::vector<vector_size_t> batchSizes = {10, 50, 100, 500};
+    bigint(true, 200, batchSizes);
+  }
+
+  void smallBigintWithPayload() {
+    const std::vector<vector_size_t> batchSizes = {10, 50, 100, 500};
+    bigint(false, 200, batchSizes);
+  }
+
+  void largeBigint() {
+    const std::vector<vector_size_t> batchSizes = {
+        1'000, 10'000, 100'000, 1'000'000};
+    bigint(true, 1, batchSizes);
+  }
+
+  void largeBigintWithPayloads() {
+    const std::vector<vector_size_t> batchSizes = {
+        1'000, 10'000, 100'000, 1'000'000};
+    bigint(false, 1, batchSizes);
+  }
+
+  void largeVarchar() {
+    const std::vector<vector_size_t> batchSizes = {
+        1'000, 10'000, 100'000, 1'000'000};
+    std::vector<RowTypePtr> rowTypes =
+        OrderByBenchmarkUtil::largeVarcharRowTypes();
+    std::vector<int> numKeys = {1, 2, 3, 4};
+    benchmark("no-payloads", "varchar", batchSizes, rowTypes, numKeys, 1);
+  }
+
+ private:
+  void benchmark(
+      const std::string& prefix,
+      const std::string& keyName,
+      const std::vector<vector_size_t>& batchSizes,
+      const std::vector<RowTypePtr>& rowTypes,
+      const std::vector<int>& numKeys,
+      int32_t numVectors) {
+    for (auto batchSize : batchSizes) {
+      for (auto i = 0; i < rowTypes.size(); ++i) {
+        const auto name = fmt::format(
+            "{}_{}_{}_{}k", prefix, numKeys[i], keyName, batchSize / 1000.0);
+        addBenchmark(name, batchSize, rowTypes[i], numVectors, numKeys[i]);
+      }
+    }
+  }
+
+  void addBenchmark(
+      const std::string& testName,
+      size_t numRows,
+      const RowTypePtr& rowType,
+      int numVectors,
+      int numKeys) {
+    auto testCase = std::make_unique<TestCase>(
+        testName, numRows, numVectors, rowType, numKeys);
+    {
+      folly::addBenchmark(
+          __FILE__,
+          "OrderBy_" + testCase->testName,
+          [test = testCase.get(), this]() {
+            auto plan = makeOrderByPlan(*test);
+            run(plan);
+            return 1;
+          });
+    }
+    testCases_.push_back(std::move(testCase));
+  }
+
+  void bigint(
+      bool noPayload,
+      int numVectors,
+      const std::vector<vector_size_t>& batchSizes) {
+    std::vector<RowTypePtr> rowTypes =
+        OrderByBenchmarkUtil::bigintRowTypes(noPayload);
+    std::vector<int> numKeys = {1, 2, 3, 4};
+    benchmark(
+        noPayload ? "no-payload" : "payload",
+        "bigint",
+        batchSizes,
+        rowTypes,
+        numKeys,
+        numVectors);
+  }
+
+  core::PlanNodePtr makeOrderByPlan(TestCase& test) {
+    folly::BenchmarkSuspender suspender;
+    std::vector<RowVectorPtr> vectors;
+    for (auto i = 0; i < test.numVectors; ++i) {
+      vectors.emplace_back(OrderByBenchmarkUtil::fuzzRows(
+          test.rowType, test.numRows, test.numKeys, pool_.get()));
+    }
+    return makeOrderByPlan(vectors, makeOrderByKeys(test.numKeys));
+  }
+
+  core::PlanNodePtr makeOrderByPlan(
```

Comment:
Maybe combine the two makeOrderByPlan functions and makeOrderByKeys function because it is used only once.